onnxruntime/core/providers/cpu/ml/ml_common.h

```diff
@@ -285,40 +285,50 @@ static inline void ComputeSoftmaxZero(std::vector<float>& values) {
 }
 
 static inline void write_scores(std::vector<float>& scores, POST_EVAL_TRANSFORM post_transform, int64_t write_index, Tensor* Z, int add_second_class) {
-  if (post_transform == POST_EVAL_TRANSFORM::PROBIT && scores.size() == 1) {
-    scores[0] = ComputeProbit(scores[0]);
-    Z->template MutableData<float>()[write_index] = scores[0];
-  } else if (scores.size() >= 2) {  //multiclass
-    if (post_transform == POST_EVAL_TRANSFORM::LOGISTIC) {
-      for (float& score : scores) {
-        score = ComputeLogistic(score);
-      }
-    } else if (post_transform == POST_EVAL_TRANSFORM::SOFTMAX) {
-      ComputeSoftmax(scores);
-    } else if (post_transform == POST_EVAL_TRANSFORM::SOFTMAX_ZERO) {
-      ComputeSoftmaxZero(scores);
+  if (scores.size() >= 2) {  //multiclass
+    switch ((unsigned char)post_transform) {
+      case (unsigned char)POST_EVAL_TRANSFORM::LOGISTIC:
+        for (float& score : scores)
+          score = ComputeLogistic(score);
+        break;
+      case (unsigned char)POST_EVAL_TRANSFORM::SOFTMAX:
+        ComputeSoftmax(scores);
+        break;
+      case (unsigned char)POST_EVAL_TRANSFORM::SOFTMAX_ZERO:
+        ComputeSoftmaxZero(scores);
+        break;
     }
-  } else {                                              //binary case
-    if (add_second_class == 0 && scores.size() == 1) {  //0=all positive weights, winning class is positive
-      scores.push_back(scores[0]);
-      scores[0] = 1.f - scores[0];                             //put opposite score in positive slot
-    } else if (add_second_class == 1 && scores.size() == 1) {  //1 = all positive weights, winning class is negative
-      scores.push_back(scores[0]);
-      scores[0] = 1.f - scores[0];                             //put opposite score in positive slot
-    } else if (add_second_class == 2 && scores.size() == 1) {  //2 = mixed weights, winning class is positive
-      if (post_transform == POST_EVAL_TRANSFORM::LOGISTIC) {
-        scores.push_back(ComputeLogistic(scores[0]));
-        scores[0] = ComputeLogistic(-scores[0]);
-      } else {
-        scores.push_back(scores[0]);
-        scores[0] = -scores[0];
-      }
-    } else if (add_second_class == 3 && scores.size() == 1) {  //3 = mixed weights, winning class is negative
-      if (post_transform == POST_EVAL_TRANSFORM::LOGISTIC) {
-        scores.push_back(ComputeLogistic(scores[0]));
-        scores[0] = ComputeLogistic(-scores[0]);
-      } else {
-        scores.push_back(-scores[0]);
+  } else if (scores.size() == 1) {  //binary case
+    if (post_transform == POST_EVAL_TRANSFORM::PROBIT) {
```

Comment:
The Probit transform should be at the same code block as Logistic, softmax and softmax_zero. It's not limited only to scores.size()==1. Can we move up? -- not sure why it is separated from the rest of the functions here.

Reply:
It is: https://github.com/Microsoft/onnxruntime/blob/xadupre/svm/onnxruntime/core/providers/cpu/ml/ml_common.h#L290.